spec/amber/pipe/params_spec.cr

```diff
@@ -0,0 +1,92 @@
+require "../../spec_helper"
+
+module Amber
+  module Pipe
+    describe Params do
+      it "parses query params" do
+        request = HTTP::Request.new("GET", "/?test=test")
+        context = create_context(request)
+        params = Params.instance
+
+        params.call(context)
+
+        context.params["test"].should eq "test"
+      end
+
+      it "parses multiple query params" do
+        request = HTTP::Request.new("GET", "/?test=test&test2=test2")
+        context = create_context(request)
+        params = Params.instance
+
+        params.call(context)
+
+        context.params["test2"].should eq "test2"
+      end
+
+      it "parses body params" do
+        headers = HTTP::Headers.new
+        headers["Content-Type"] = "application/x-www-form-urlencoded"
+        body = "test=test"
+        request = HTTP::Request.new("POST", "/", headers, body)
+        context = create_context(request)
+        params = Params.instance
+
+        params.call(context)
+
+        context.params["test"].should eq "test"
+      end
+
+      it "parses body params with charset" do
+        headers = HTTP::Headers.new
+        headers["Content-Type"] = "application/x-www-form-urlencoded; charset=UTF-8"
+        body = "test=test"
```

Comment:
Can you give me more context about the utf-8 chars? Do you mean something like test=\x3D?